### ISSUE TITLE
Proposal: Optionally return a document ID on valueChanges()

### ISSUE THREAD

**Issue by codediodeio** [angular/angularfire#1973]
A major pain point with angularfire continues to be mapping a document ID to a collection query. I propose we add an optional `withId` argument to `collection.valueChanges()`. This can be a simple non-breaking change and I can PR it. 

```js
collection.valueChanges(withId?: string)

// emits [ { myIdKey: 'MrfFpRBfWLTd7LqiTt9u', ...data }, ... ]
```

This will greatly simplify the code below

```js
// current requirement
ref.snapshotChanges()
      .pipe(
        map(actions => {
          return actions.map(a => {
            const data = a.payload.doc.data();
            const id = a.payload.doc.id;
            return { myIdKey: id, ...data };
          });
})

// after change 

ref.valueChanges('myIDKey')
```

**Comment by AJHanekom** [comment]
Yes please.

**Comment by balajkhan07** [comment]
That would be great.

**Comment by rtpHarry** [comment]
This can be closed it was merged with #1976 (and thanks)
